File: backend/models/feedback.go
package models

import "time"

const (
	FeedbackStatusOpen   = "open"
	FeedbackStatusClosed = "closed"
)

type Feedback struct {
	ID            uint       `json:"id" gorm:"primaryKey"`
	UserID        uint       `json:"user_id" gorm:"not null;index"`
	ScreenName    string     `json:"screen_name" gorm:"size:128;not null;default:'';index"`
	Message       string     `json:"message" gorm:"not null;type:text"`
	Status        string     `json:"status" gorm:"size:16;not null;default:open;index"`
	AdminResponse string     `json:"admin_response" gorm:"type:text"`
	RespondedAt   *time.Time `json:"responded_at"`
	RespondedBy   *uint      `json:"responded_by"`
	CreatedAt     time.Time  `json:"created_at"`
	UpdatedAt     time.Time  `json:"updated_at"`
}

File: frontend/lib/models/feedback.dart
class FeedbackItem {
  final int id;
  final int userId;
  final String screenName;
  final String message;
  final String status;
  final String adminResponse;
  final DateTime? respondedAt;
  final int? respondedBy;
  final DateTime createdAt;
  final String? username;

  const FeedbackItem({
    required this.id,
    required this.userId,
    required this.screenName,
    required this.message,
    required this.status,
    required this.adminResponse,
    this.respondedAt,
    this.respondedBy,
    required this.createdAt,
    this.username,
  });

  bool get isOpen => status == 'open';
  bool get hasResponse => adminResponse.trim().isNotEmpty;

  factory FeedbackItem.fromJson(Map<String, dynamic> json) {
    return FeedbackItem(
      id: (json['id'] as num).toInt(),
      userId: (json['user_id'] as num).toInt(),
      screenName: json['screen_name'] as String? ?? '',
      message: json['message'] as String? ?? '',
      status: json['status'] as String? ?? 'open',
      adminResponse: json['admin_response'] as String? ?? '',
      respondedAt: _parseDate(json['responded_at']),
      respondedBy: (json['responded_by'] as num?)?.toInt(),
      createdAt: _parseDate(json['created_at']) ?? DateTime.now(),
      username: json['username'] as String?,
    );
  }

  static DateTime? _parseDate(dynamic raw) {
    if (raw == null) return null;
    if (raw is String && raw.isNotEmpty) {
      return DateTime.tryParse(raw);
    }
    return null;
  }
}

class FeedbackPage {
  final List<FeedbackItem> items;
  final int total;
  final int page;
  final int pageSize;

  const FeedbackPage({
    required this.items,
    required this.total,
    required this.page,
    required this.pageSize,
  });

  int get totalPages {
    if (total == 0) return 1;
    return ((total - 1) ~/ pageSize) + 1;
  }
}
